Extract function from ctor of Ledger

hewison-chris opened the pull request:
Adding `updateStateFromStorageBlock` as protected function in Ledger
constructor. This helps with unit testing of the ledger where we want to
add preimages after the enrollments have been loaded into state.

hewison-chris committed:
Extract function from constructor of Ledger

Adding `replayStoredBlock` as protected function in Ledger.
This helps with unit testing of the ledger where we want to
add preimages after the enrollments have been loaded into state.

linked0 commented on source/agora/node/Ledger.d:
```diff
@@ -432,6 +424,26 @@ public class Ledger
+    {
+        // Make sure our data on disk is valid
+        if (auto fail_reason = this.validateBlock(block))
+            throw new Exception(
+                "A block loaded from disk is invalid: " ~
+                fail_reason);
```
```suggestion
            throw new Exception("A block loaded from disk is invalid: " ~
                fail_reason);
```

linked0 reviewed (approved):
LGTM

kchulj reviewed (approved):
LG

hewison-chris commented:
System test failed. Triggered tests to run again.

Geod24 commented on source/agora/node/Ledger.d:
```diff
@@ -432,6 +424,25 @@ public class Ledger
+
+    ***************************************************************************/
+
+    protected void updateStateFromStorageBlock (in Block block) @safe
```
I would think that encapsulating the whole `if` would be better, unless you have other plans ?
E.g.
```D
if (this.utxo_set.length == 0)
    this.updateStateFromStorageBlock();
```

Also... `this.replayBlockchain()` or something a bit more succinct ? :D

hewison-chris replied:
I need to have it called per block but I can rename it.

Geod24 commented on source/agora/node/Ledger.d:
```diff
@@ -188,15 +188,7 @@ public class Ledger
-                        fail_reason);
-
-                this.addValidatedBlock(block);
+                updateStateFromStorageBlock(this.storage.readBlock(Height(height)));
```
```suggestion
                this.updateStateFromStorageBlock(this.storage.readBlock(Height(height)));
```

Geod24 reviewed (approved):
Just style nits

hewison-chris committed:
More retries for system test nodes

We are having problems with the nodes getting banned during the system
integration test. This is an attempt to fix it by increasing retries of
rest calls to the node.

hewison-chris commented:
Added a commit with 50 retries for nodes in system test. Let's see if it goes green 🤞
